File: DOHEE/Week1/기사 단원의 무기.py
# 30분
import math

def solution(number, limit, power):
    answer = 0
    for i in range(1, number + 1):
        origin_power = []
        for j in range(1, int(i ** 0.5) + 1):
            if i % j == 0:
                origin_power.append(j)
                if math.sqrt(i) != j:
                       origin_power.append(i / j)
        if len(origin_power) > limit:
            answer += power
        else:
            answer += len(origin_power)
    return answer

File: DOHEE/Week1/기사 단원의 무기.js
// 40분
function solution(number, limit, power) {
    var answer = 0;
    for (let i = 1; i < number + 1; i++) {
        let originPower = 0;
        for (let j = 1; j <= i / 2; j++) {
            if (i % j === 0) originPower += 1;
        }
        answer += originPower + 1 > limit ? power : originPower + 1;
    }
    return answer;
}
